### PR TITLE
K1J-511: Update to java 21, gradle 8.10.2, Spring Framework 6.1, remove gradle-intyg-plugin

### PR DESCRIPTION
**Pull request by mikeyOne**

### CODE REVIEW

File: Jenkins.properties [hunk 1]
```diff
@@ -1,7 +1,7 @@
 project.name=intyg
 
 artifact.name=intygsadmin
-artifact.version=1.10.0.4
+artifact.version=1.11.0.0
```
Comment (mww-sopra-steria) [inline Jenkins.properties:4]: Reminder that we update builder and runtime images

File: persistence/src/main/java/se/inera/intyg/intygsadmin/persistence/entity/BannerEntity.java [hunk 2]
```diff
@@ -44,7 +45,7 @@ public class BannerEntity {
 
     @Id
     @GeneratedValue(strategy = GenerationType.AUTO)
-    @Type(type = "uuid-char")
+    @JdbcTypeCode(SqlTypes.CHAR)
```
Comment (mww-sopra-steria) [inline persistence/src/main/java/se/inera/intyg/intygsadmin/persistence/entity/BannerEntity.java:48]: Would like to understand the usage of this as I haven't seen it before. Not that I doubt it, just of curiosity.

File: web/src/main/java/se/inera/intyg/intygsadmin/web/auth/AuthenticationConstansts.java [hunk 3]
```diff
@@ -20,13 +20,13 @@
 
 public final class AuthenticationConstansts {
 
-    public static final String LOGIN_URL = "/login";
+    public static final String LOGIN_URL = "/oauth2/authorization/siths";
+    public static final String LOGIN_REDIRECT_URL = "/login/inera";
     public static final String LOGOUT_URL = "/logout";
     public static final String FAKE_PROFILE = "fake";
-    public static final String FAKE_LOGIN_ENDPOINT = "/fake";
+    public static final String FAKE_LOGIN_ENDPOINT = "/fake-api/login";
```
Comment (mww-sopra-steria) [inline web/src/main/java/se/inera/intyg/intygsadmin/web/auth/AuthenticationConstansts.java:27]: Not that it matters much but should we "put the fake login" in testability. /fake-api/login -> /testability/fake (or /testablility/login)

File: web/src/main/java/se/inera/intyg/intygsadmin/web/auth/IntygsadminUser.java [hunk 4]
```diff
@@ -44,4 +50,25 @@ public IntygsadminUser(UserEntity userEntity, AuthenticationMethod authenticatio
         this.name = userEntity.getName();
     }
 
+    @Override
```
Comment (mww-sopra-steria) [inline web/src/main/java/se/inera/intyg/intygsadmin/web/auth/IntygsadminUser.java:53]: We can use Lombok annotations for providing implementation of equals and hashcode

File: web/src/main/java/se/inera/intyg/intygsadmin/web/config/SecurityConfig.java [hunk 5]
```diff
@@ -20,238 +20,205 @@
...
-import se.inera.intyg.intygsadmin.web.auth.filter.IndividualClaimsOuth2ContextFilter;
-import se.inera.intyg.intygsadmin.web.auth.filter.IneraOidcFilter;
+import se.inera.intyg.intygsadmin.web.auth.IntygsadminUser;
 import se.inera.intyg.intygsadmin.web.auth.filter.SessionTimeoutFilter;
-import se.inera.intyg.intygsadmin.web.service.monitoring.MonitoringLogServiceImpl;
+import se.inera.intyg.intygsadmin.web.service.monitoring.MonitoringLogService;
 
 @Configuration
 @EnableWebSecurity
+@EnableMethodSecurity
```
Comment (mww-sopra-steria) [inline web/src/main/java/se/inera/intyg/intygsadmin/web/config/SecurityConfig.java:70]: How does EnableMethodSecurity work and why do we need it?

**Review by mww-sopra-steria — COMMENTED** [review]
Looks very good and I don't have anything in particular that needs to be looked into. But there are a few things I'm curious about and would like to know more about. 

Could we go through them together?

**Review by mww-sopra-steria — APPROVED** [review]
Great job! 💪

File: web/src/main/java/se/inera/intyg/intygsadmin/web/auth/AuthenticationConstansts.java (hunk 3 above)
Comment (mikaelaromild1) [reply web/src/main/java/se/inera/intyg/intygsadmin/web/auth/AuthenticationConstansts.java:27]: We have decided to not change it at this time.

**Pull request merged**